persist/fs/msgpack/decoder.go

```diff
@@ -360,47 +374,55 @@ func (dec *Decoder) decodeLogMetadata() schema.LogMetadata {
 	return logMetadata
 }
 
-func (dec *Decoder) decodeRootObject(expectedVersion int, expectedType objectType) int {
-	dec.checkVersion(expectedVersion)
+func (dec *Decoder) decodeRootObject(expectedVersion int, expectedType objectType) (version int, numFieldsToSkip int) {
+	version = dec.checkVersion(expectedVersion)
 	if dec.err != nil {
-		return 0
+		return 0, 0
 	}
-	numFieldsToSkip, ok := dec.checkNumFieldsFor(rootObjectType)
+	numFieldsToSkip, _, ok := dec.checkNumFieldsFor(rootObjectType)
 	if !ok {
-		return 0
+		return 0, 0
 	}
 	actualType := dec.decodeObjectType()
 	if dec.err != nil {
-		return 0
+		return 0, 0
 	}
 	if expectedType != actualType {
 		dec.err = fmt.Errorf("object type mismatch: expected %v actual %v", expectedType, actualType)
-		return 0
+		return 0, 0
 	}
-	return numFieldsToSkip
+	return version, numFieldsToSkip
 }
 
-func (dec *Decoder) checkVersion(expected int) {
+func (dec *Decoder) checkVersion(expected int) int {
 	version := int(dec.decodeVarint())
 	if dec.err != nil {
-		return
+		return 0
 	}
 	if version > expected {
 		dec.err = fmt.Errorf("version mismatch: expected %v actual %v", expected, version)
+		return 0
 	}
+
+	return version
 }
 
-func (dec *Decoder) checkNumFieldsFor(objType objectType) (int, bool) {
+func (dec *Decoder) checkNumFieldsFor(objType objectType) (int, int, bool) {
 	actual := dec.decodeNumObjectFields()
 	if dec.err != nil {
-		return 0, false
+		return 0, 0, false
 	}
-	expected := numFieldsForType(objType)
-	if expected > actual {
-		dec.err = fmt.Errorf("number of fields mismatch: expected %d actual %d", expected, actual)
-		return 0, false
+	min, curr := numFieldsForType(objType)
+	if min > actual {
+		dec.err = fmt.Errorf("number of fields mismatch: expected minimum of %d actual %d", min, actual)
+		return 0, 0, false
+	}
+
+	numToSkip := actual - curr
```

Comment:
could you put a little blurb explaining why this is safe to do

Reply:
Not totally following, what kind of comment would be helpful here?